### PR TITLE
Rename top-level view components

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR renames several components to use a new `*View` naming convention for top-/route-level view components. Root component `HypothesisApp` renders one of these `*View` components depending on route.

This naming change was discussed in Slack.

New Notebook component will be named `NotebookView` to follow suit.

### CODE REVIEW

**Review by esanzgar — APPROVED** [review]
👍🏼

File: src/sidebar/components/sidebar-view.js [hunk 1]
```diff
@@ -151,18 +151,14 @@ function SidebarContent({
   );
 }
 
-SidebarContent.propTypes = {
+SidebarView.propTypes = {
```
Comment (esanzgar) [inline src/sidebar/components/sidebar-view.js:154]: Given that types are provided via TS-JSDocs, isn't this redundant?

File: src/sidebar/components/stream-view.js [hunk 2]
```diff
@@ -82,11 +82,11 @@ function StreamContent({ api, toastMessenger }) {
   return <ThreadList thread={rootThread} />;
 }
 
-StreamContent.propTypes = {
+StreamView.propTypes = {
```
Comment (esanzgar) [inline src/sidebar/components/stream-view.js:85]: Idem as above.

File: src/sidebar/components/annotation-view.js [hunk 3]
```diff
@@ -98,11 +98,11 @@ function AnnotationViewerContent({ loadAnnotationsService, onLogin }) {
   );
 }
 
-AnnotationViewerContent.propTypes = {
+AnnotationView.propTypes = {
```
Comment (esanzgar) [inline src/sidebar/components/annotation-view.js:101]: Given that types are provided via TS-JSDocs, isn't this redundant?

File: src/sidebar/components/sidebar-view.js (hunk 1 above)
Comment (lyzadanger) [reply src/sidebar/components/sidebar-view.js:154]: You’re right that this is rather redundant! There is some history and context to the current structure you see here. Before we introduced type checking in JSDoc comments, we relied on `propTypes` for various type-checking and type-checking-like functionality in components. `propTypes` also provides _run-time_ validation and warnings, which can be handy when developing. 

But we’ve already identified that doing both is a little much. And Rob already removed `propTypes` from `lms`:  https://github.com/hypothesis/lms/pull/1952

As I understand it, the plan is to ultimately remove `propTypes` from the `client` codebase as well. We have a few things to do before we can do that. For example, the `Button` component relies on `propTypes`’  custom validator function feature: https://github.com/hypothesis/client/blob/dea9b141459932c5065f4c96d9d626622d2d249f/src/sidebar/components/button.js#L112

**Pull request merged**